lib/mgmt_msg_native.h

```diff
@@ -152,6 +152,21 @@ DECLARE_MTYPE(MSG_NATIVE_GET_DATA);
 #define MGMT_MSG_CODE_TREE_DATA 2
 #define MGMT_MSG_CODE_GET_DATA	3
 
+/*
+ * Datastores
+ */
+#define MGMT_MSG_DATASTORE_STARTUP     0
+#define MGMT_MSG_DATASTORE_CANDIDATE   1
+#define MGMT_MSG_DATASTORE_RUNNING     2
+#define MGMT_MSG_DATASTORE_OPERATIONAL 3
+
+/*
+ * Formats
+ */
+#define MGMT_MSG_FORMAT_XML    0
+#define MGMT_MSG_FORMAT_JSON   1
+#define MGMT_MSG_FORMAT_BINARY 2 /* non-standard libyang internal format */
```

Comment:
I don't agree with changing away from LYD_FORMAT. This is introducing an unnecessary indirection and possible diverging errors. Now whenever we need to pass a LYD_FORMAT we have to do a mapping, or assume they are the same. I don't want to build assumptions into the code -- or unneeded mappings.

Reply:
mgmtd frontend API is a public API intended to be used by people for configuring FRR from scripts/tools/etc. We can't ensure backwards compatibility if we rely on libyang-defined variables. What happens if libyang decides to change these values in a new major release? All user scripts would just break. We can use LYD_FORMAT between mgmtd and backends, but not on the frontend.

Reply:
Also, when someone external to FRR developments wants to write a script to configure FRR through mgmtd, how do they even know what LYD_FORMAT is? Do they need to download libyang sources to figure out which number to set?

Reply:
I didn't notice that this was exclusively for the frontend. Your argument makes sense.

Reply:
I changed the values to use LYD_FORMAT values directly and added `_Static_assert`s.